### ISSUE TITLE
Allow properties to be only a single value but able to be any in an array

### ISSUE THREAD

**Issue by kettek** [kettek/schisma#2]
In short, when create() or other methods are called, schema properties that are of type Array but have a "singleValue" or similar flag should populate/validate that property as any of the types in Array.

In other words:
```
let MyScheme = schisma({
  myProp: {
    $type: [String, Number],
    $singleValue: true,
  }
})

MyScheme.validate({
  myProp: 32
})
// => valid!

MyScheme.validate({
  myProp: "32!"
})
// => valid!

MyScheme.validate({
  myProp: [55]
})
// => invalid!
```

**Comment by kettek** [comment]
Maybe "$singleType"

**Comment by kettek** [comment]
Potentials:
  * $notArray
  * $singleType
  * $matchOnce
  * $matchOneType
  * $matchSingleType
  * $singleton

**Comment by kettek** [comment]
$typeof seems to be nice.

Potentially:

```
let MyScheme = schisma({
  myProp: {
    $typeof: [String, Number]
  }
})
```

**Comment by kettek** [comment]
For comparison, an array of types would be:

```
let MyScheme2 = schisma({
  myProp: {
    $type: [String, Number]
  }
})
```

**Comment by kettek** [comment]
Fixed in 8c8f72a281aeee38534a39934da2aad236f6f527 !
